diff --git a/provider/provider.go b/provider/provider.go
index 579d430..1a73496 100644
--- a/provider/provider.go
+++ b/provider/provider.go
@@ -86,6 +86,10 @@ func getDefaultPath() string {
 		//Hardcoding should be fine for now since SUMMON_PROVIDER and -p are available
 		return "C:\\Program Files\\Cyberark Conjur\\Summon\\Providers"
 	} else {
-		return "/usr/local/lib/summon"
+		value, exists := os.LookupEnv("SUMMON_PROVIDER_DIR")
+		if !exists {
+			value = "/usr/local/lib/summon"
+		}
+		return value
 	}
 }
